### ISSUE TITLE
In Manual Optimization mode, loss does not converge when the Train Batch Size is 1

### ISSUE THREAD

**Issue by OPilgrim** [Lightning-AI/lightning#14527]
### Discussed in https://github.com/Lightning-AI/lightning/discussions/14526

<div type='discussions-op-text'>

<sup>Originally posted by **OPilgrim** September  4, 2022</sup>
This is very confusing that if the train batch size is set to 1 and Automatic Optimization mode is used, accumulate_grad_batches=32, train loss can converge normally. However, using Manual Optimization mode with automatic_optimization=False, the train loss fails to converge. Is there any other code I'm ignoring in PyTorch Lightning?
 
The code used is as follows:
     def training_step(self, batch, batch_idx, optimizer_idx=None):
            opt = self.optimizers()
            loss = gen_helper()
            self.log('train/loss', loss, prog_bar=True, logger=True)
            self.manual_backward(loss)
            if (batch_idx+1) % self.hparams.accumulate_grad_batches == 0:
                opt.step()
                opt.zero_grad()</div>
